lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2582,69 +2609,77 @@ void TFDeabstraction::doIt() {
   maybeLogFuncSize(fn, "AfterDA");
 }
 
-
 namespace {
-  struct TFDeabstractionPass : public SILModuleTransform {
-    /// The entry point to the transformation, runs deabstraction on an entire
-    /// module.
-    void run() override;
-  };
-}  // end anonymous namespace
 
-void TFDeabstractionPass::run() {
-  SILModule *module = getModule();
-  auto &ctx = module->getASTContext();
+/// A helper class to launch deabstraction on functions.
+class TFDeabstractionHelper {
+public:
+  TFDeabstractionHelper(SILTransform &transform, SILModule *module)
+      : transform(transform), module(module), constantEvaluator(*module) {}
+
+  /// Deabstract the given function and returns true if this function was
+  /// deabstracted. If the flag forceTFFunctions is true, forces partitioning of
+  /// functions that operate on Tensors even if it would have been rejected
+  /// otherwise.
+  bool deabstract(SILFunction &fn, bool forceTFFunctions);
```

Comment:
can `deabstract` be a private method instead?

Reply:
Done.